internal/certgen/certgen.go

```diff
@@ -28,7 +28,16 @@ import (
 	"io/ioutil"
 	"math/big"
 	"os"
+	"path"
 	"time"
+
+	corev1 "k8s.io/api/core/v1"
```

Comment:
It's time to start splitting these files up. This is for a variety of reasons. The first is obviously length, but it's not just an arbitrary file length, its more than this file contains various operations that are related but independant in their use and operation.

For example we have functions that make TLS certificates, others that write out certificates to a file system, and some that create a k8s client to write certificates to k8s.

The hint is when a file called `certgen.go` grows imports required to write things to k8s, you know its doing too many things.

I'd suggest organising the functions in this file along either functional boundaries; eg, certs.go for things that generate certs, k8s.go for things that talk to k8s, and so on -- or along import boundaries, eg. group functions by the imports they require -- generally this has the same result.